src/menus/Menu.java

```diff
@@ -0,0 +1,27 @@
+package menus;
+
+import DLibX.*;
+import utils.Gamepad;
+import java.util.ArrayList;
+import net.java.games.input.Controller;
+
+public abstract class Menu {
+
+    protected DConsole dc;
+    protected ArrayList<Gamepad> gamepads;
+    protected Controller[] controllers;
+
+    public Menu(DConsole dc, ArrayList<Gamepad> gamepads, Controller[] controllers) {
```

Comment:
I feel like it's not necessary to pass in the controllers for a total of 3 lines of code. Controller polling should probably be done always if we want to use it as the main control scheme

Reply:
but polling has done be done through controller and input checking has to be done through gamepads no?

Reply:
what are you suggesting?

Reply:
What I mean is controller polling can be done in the main while loop instead of per-menu

Reply:
okay lol i see my issue now. I was doing this before and it didn't seem to work because it wouldn't exit because it was just one game and polling in the loop in App. I think we will need another screen to test it properly.

Reply:
All good, the functionality is correct and this is just a nitpick so I'll just approve

Reply:
After some playing around here is the issue. If we have a while loop in the game that doesn't poll controllers, the game can never actually take input because the for loop in the App() while loop that polls controllers doesn't actually get input since the game while loop is running and not polling. So the way I see it we eaither think of a different way to exit games or keep it the way it is. LMK if im missing somthing.

Reply:
I think the way we exit games right now is probably fine. It'd be a lot of effort for something so minuscule so I think its best to just let there be boilerplate rather than devise a whole new plan for exiting